### ISSUE TITLE
`lr`, `momentum`, `weight_decay` and `dampening` parameter of `optim.SGD()` work with `bool` values

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#137367]
### 🐛 Describe the bug

[The doc](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html) of `optim.SGD()` doesn't say that the type of `lr`, `momentum`, `weight_decay` and `dampening` parameter are `bool` as shown below:

> Parameters
> - ...
> - lr ([float](https://docs.python.org/3/library/functions.html#float), optional) – learning rate (default: 1e-3)
> - momentum ([float](https://docs.python.org/3/library/functions.html#float), optional) – momentum factor (default: 0)
> - weight_decay ([float](https://docs.python.org/3/library/functions.html#float), optional) – weight decay (L2 penalty) (default: 0)
> - dampening ([float](https://docs.python.org/3/library/functions.html#float), optional) – dampening for momentum (default: 0)
> - ...

But `lr`, `momentum`, `weight_decay` and `dampening` parameter work with `bool` values as shown below:

```python
from torch import nn
from torch import optim

class MyModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear_layer = nn.Linear(in_features=4, out_features=5)
    
    def forward(self, x):
        return self.linear_layer(x)

mymodel = MyModel()

sgd = optim.SGD(params=mymodel.parameters(), lr=True, momentum=True, 
                dampening=True, weight_decay=True)
sgd
# SGD (
# Parameter Group 0
#     dampening: True
#     differentiable: False
#     foreach: None
#     fused: None
#     lr: True
#     maximize: False
#     momentum: True
#     nesterov: False
#     weight_decay: True
# )
```

### Versions

```python
import torch

torch.__version__ # '2.3.0'
```

**Comment by ivnvalex** [comment]
I would suggest to ```raise ValueError``` in case ```bool``` is passed
I think I can handle this if it is not taken yet and there are no concerns
Can you please assign me?
Thanks

**Comment by janeyx99** [comment]
See my comment here: https://github.com/pytorch/pytorch/issues/137373#issuecomment-2407915656
